src/NuGet.Services.EndToEnd/Support/Package.cs

```diff
@@ -61,6 +72,30 @@ public static Package Create(PackageCreationContext context)
             return new Package(context.Id, context.NormalizedVersion, context.FullVersion, nupkgBytes);
         }
 
+        public static Package Create(PackageCreationContext context, IEnumerable<string> files)
+        {
+            return Create(context, files, PackageProperties.Default());
+        }
+
+        public static Package Create(PackageCreationContext context, IEnumerable<string> files, PackageProperties properties)
+        {
+            var physicalPackageFilesList = files
+                .Select(x => new { FileName = Path.GetFileName(x), Stream = GetMemoryStreamForFile(x) })
+                .Select(x => new PhysicalPackageFile(x.Stream) {
+                    TargetPath = $"lib/{x.FileName}"
+                });
+
+            ReadOnlyCollection<byte> nupkgBytes;
+            using (var nupkgStream = TestData.BuildPackageStreamForFiles(context, physicalPackageFilesList, properties.IsSymbolsPackage))
+            {
+                var bufferStream = new MemoryStream();
+                nupkgStream.CopyTo(bufferStream);
+                nupkgBytes = Array.AsReadOnly(bufferStream.ToArray());
```

Comment:
You don't need to do this. `ToArray()` copies the internal buffer.

Reply:
caller needs readonly protection.